File: src/test/java/com/kasakaid/kasakaidboot/configuration/ApplicationPropertiesTest.java
package com.kasakaid.kasakaidboot.configuration;

import com.kasakaid.kasakaidboot.Properties;
import com.kasakaid.kasakaidboot.helper.AbstractBaseTest;
import lombok.extern.slf4j.Slf4j;
import org.junit.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.Matchers.nullValue;

@Slf4j
public class ApplicationPropertiesTest extends AbstractBaseTest {

    @Test
    public void applicationPropertiesの読み込み() {
        assertThat(Properties.getProperty("common.key"), is("this_is_common"));
    }

    @Test
    public void テストのみのキーのテスト() {
        assertThat(Properties.getProperty("test.key"), is("this_is_test_key"));
    }

//    @Test(expected = IllegalArgumentException.class)
    @Test
    public void 開発環境のキーは取得できないテスト() {
        assertThat(Properties.getProperty("development.key"), nullValue());
    }
}

File: src/test/kotlin/com/kasakaid/boot/configuration/ApplicationPropertiesTest.kt
package com.kasakaid.boot.configuration

import com.kasakaid.boot.AbstractBaseTest
import org.hamcrest.CoreMatchers.nullValue
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.core.Is.`is`
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.core.env.Environment

class ApplicationPropertiesTest : AbstractBaseTest() {

    @Autowired
    lateinit var env: Environment

    @Test
    fun applicationPropertiesの読み込み() {
        assertThat(env.getProperty("common.key"), `is`("this_is_common"))
    }

    @Test
    fun テストのみのキーのテスト() {
        assertThat(env.getProperty("include.test"), `is`("this_is_test_key"))
    }

    @Test
    fun testのプロパティファイルにのみ記述されている内容を取得できる() {
        assertThat(env.getProperty("include.test"), `is`("this_is_test_key"))
    }

    @Test
    fun プロファイルと同じ滑のファイルのキーを取得できる() {
        assertThat(env.getProperty("duplicate.key"), `is`("I'm a test"))
    }

    @Test
    fun developmentのキーはnullになる() {
        assertThat(env.getProperty("include1.development"), `is`(nullValue()))
        assertThat(env.getProperty("include2.development"), `is`(nullValue()))
    }

    //    @Test(expected = IllegalArgumentException.class)
    @Test
    fun 開発環境のキーは取得できないテスト() {
        assertThat(env.getProperty("development.key"), nullValue())
    }
}